src/apps/tools.clj

```diff
@@ -96,19 +108,24 @@
   [{:keys [tools]}]
   (transaction
     (let [tool-ids (doall (map add-new-tool tools))]
+      (dorun (map permissions/register-public-tool tool-ids))
       {:tool_ids tool-ids})))
 
 (defn update-tool
-  [overwrite-public {:keys [id container] :as tool}]
+  [user overwrite-public {:keys [id container] :as tool}]
   (persistence/update-tool tool)
   (when container
     (set-tool-container id overwrite-public container))
-  (get-tool id))
+  (get-tool user id))
 
 (defn delete-tool
   [user tool-id]
-  (let [{:keys [name location version]} (get-tool tool-id)]
+  (let [{:keys [name location version]} (get-tool user tool-id)]
     (validate-tool-not-used tool-id)
     (log/warn user "deleting tool" tool-id name version "@" location))
   (delete tools (where {:id tool-id}))
+  (try+
+    (permissions/delete-tool-resource tool-id)
+    (catch [:status 404] _
+      (log/warn "tool resource" tool-id "not found by permissions service")))
   nil)
```

Comment:
Yeah, that sounds reasonable. It will be better when we can add the admin permission level. The primary reason that I hadn't added it yet was because we didn't have LDAP groups available to the DE, which meant that permissions couldn't easily be granted to all administrator groups. Now that we have LDAP groups being loaded, we'll probably want to revisit this at some point in the near future.